src/yoroi-wallets/cardano/transactionManager.ts

```diff
@@ -0,0 +1,78 @@
+import {YoroiStorage} from '../storage'
+import {BackendConfig, Transactions} from '../types'
+import {TransactionCache} from './shelley'
+
+export const makeTransactionManager = async (storage: YoroiStorage, backendConfig: BackendConfig) => {
+  const transactionCache = await TransactionCache.create(storage.join('txs/'))
+  const memosManager = await makeMemosManager(storage.join('memos/'))
+
+  return {
+    // transactionCache api
+    getTransactions() {
+      const memos = memosManager.getMemos()
+      return Object.entries(transactionCache.transactions).reduce(
+        (result, [txId, tx]) => ({
+          ...result,
+          [txId]: {
+            ...tx,
+            memo: memos[txId] ?? null,
+          },
+        }),
+        {} as Transactions,
+      )
+    },
+    getPerRewardAddressCertificates() {
+      // to get the updates
+      return transactionCache.perRewardAddressCertificates
+    },
+    getPerAddressTxs() {
+      // to get the updates
```

Comment:
are you going to these?

Reply:
I tested that https://github.com/Emurgo/yoroi-mobile/pull/2306#discussion_r1091809650